addons/test_mail/tests/test_mail_mail.py

```diff
@@ -132,6 +133,25 @@ def test_mail_mail_send_server(self):
         self.assert_email_sent_smtp(message_from='user_2@test_2.com', emails_count=5, from_filter=self.server_domain_2.from_filter)
         self.assert_email_sent_smtp(message_from='user_1@test_2.com', emails_count=5, from_filter=self.server_domain.from_filter)
 
+    def test_mail_mail_schedule(self):
+        """Test that a mail scheduled in the past/future are sent or not"""
+        scheduled_dates = [False,
+                           '',
+                           'This is not a date format',
+                           (datetime.now() - timedelta(days=1)).strftime(DEFAULT_SERVER_DATETIME_FORMAT),
+                           (datetime.now() + timedelta(days=1)).strftime(DEFAULT_SERVER_DATETIME_FORMAT),
+                           (datetime.now() + timedelta(days=1)).strftime("%H:%M:%S %d-%m-%Y")]
```

Comment:
Out of curiosity, how does parser deal if we have a timezone encoded in the format ? Couldn't we add an input for this ?

Reply:
According to the documentation https://dateutil.readthedocs.io/en/stable/parser.html the parser can handle the timezone but I don't think this will impact us because it will be at the end of the string (and because we use string compare).
Example:
```
>>>from dateutil.parser import parse
>>>d = parse('2009/05/13 19:19:30 -0400')
>>>print(d)
2009-05-13 19:19:30-04:00
```

Reply:
Then a small test and hop :) because one issue we have is that models like event have templates that may send a scheduled datetime with a timezone set in it (and crac).